### DISCUSSION TITLE
Get an email address

### DISCUSSION THREAD

**Discussion by Lex-Kent** [speechly/speechly#299] [Q&A] [answered]
Hi all,

I'd like to get an email address from the user. I'm not expecting any preamble like "My email address is... " I simply want to convert what is said into an email address. I am aware of the [email_address](https://docs.speechly.com/reference/entity-data-types#email_address) data type, but I don't really know how to use it. I created what I thought would work based on the documentation, and the best I could is

![image](https://user-images.githubusercontent.com/107930534/230138257-27eb51b5-9c5c-455b-b677-35369f305fde.png)

but this doesn't seem to work at all:

![image](https://user-images.githubusercontent.com/107930534/230138505-d2be8c73-e637-4927-9a08-e8eaa140a27a.png)

Can anyone help at all please!?

I can honestly say I've never struggled so much with any documentation as I do with Speechly's!

**Comment by teelisyys** [comment, marked as answer]
Hi @Lex-Kent ! Thank you for using Speechly!

Change the _Text-Adaptation_ from your screen shot to be 

~~~
*email [john@theweb.com](email)
*email [jane@theworld.co.uk](email)
~~~

Then the entity value will contain the formatted email address. The transcript will still be written out:

<img width="376" alt="image" src="https://user-images.githubusercontent.com/9036453/230290317-049d476c-ef1e-4cf1-912b-8140a94e33a0.png">


For recognition of more generic email addresses, you might also want to try:

~~~
*email [$SPEECHLY.EMAIL_ADDRESS](email)
~~~

**Reply by Lex-Kent** [reply]
Wow, it was as easy as that! Thank you @teelisyys :o)
